File: server/core/domain/src/main/kotlin/io/github/vinicreis/pubsub/server/core/model/data/Queue.kt
package io.github.vinicreis.pubsub.server.core.model.data

import java.util.*

data class Queue(
    val id: UUID,
    val code: String,
    val name: String = code,
    val type: Type,
    var pendingMessagesCount: Long = 0,
) {
    enum class Type {
        SIMPLE,
        MULTIPLE
    }
}

File: py-client/core/domain/queue.py
import uuid
from enum import Enum


class Queue:
    class Type(Enum):
        SIMPLE = 1
        MULTIPLE = 2

    def __init__(self,
                 code: str,
                 name: str,
                 queue_type: Type,
                 guid: str = str(uuid.uuid4()),
                 pending_messages_count: int = 0):
        self.guid = guid
        self.code = code
        self.name = name
        self.queue_type = queue_type
        self.pending_messages_count = pending_messages_count

    def __str__(self):
        return (f"{self.name}\n" +
                f"\tID: {self.guid}\n" +
                f"\tCode: {self.code}\n" +
                f"\tType: {self.queue_type}\n" +
                f"\tHas {self.pending_messages_count} pending messages")
